src/wasm/wasm.cpp

```diff
@@ -932,15 +905,19 @@ void Push::finalize() {
   }
 }
 
-size_t Function::getNumParams() { return params.size(); }
+size_t Function::getNumParams() { return sig.params.size(); }
 
 size_t Function::getNumVars() { return vars.size(); }
 
-size_t Function::getNumLocals() { return params.size() + vars.size(); }
+size_t Function::getNumLocals() { return sig.params.size() + vars.size(); }
 
-bool Function::isParam(Index index) { return index < params.size(); }
+bool Function::isParam(Index index) { return index < sig.params.size(); }
 
-bool Function::isVar(Index index) { return index >= params.size(); }
+bool Function::isVar(Index index) {
+  auto base = getVarIndexBase();
+  // TODO: update me
+  return base <= index /*&& index < base + vars.size()*/;
```

Comment:
I think the eventual update here should be to make that last part an assertion - it is invalid to ask avoid an invalid index.